transcendence-app/src/auth/oauth42.strategy.ts

```diff
@@ -62,17 +67,25 @@ export class OAuth42Strategy extends PassportStrategy(Strategy, 'oauth42') {
   }
 
   async validate(accessToken: string): Promise<User | null> {
-    const { userDto, avatarUrl } = await this.api42Service.get42UserData(
-      accessToken,
-    );
-    const dbUser = await this.userService.retrieveUserWithUserName(
-      userDto.username,
+    const { userDto, avatarUrl, providerId } =
+      await this.api42Service.get42UserData(accessToken);
+    const authProvider = await this.authProviderService.retrieveProvider(
+      AuthProviderType.FortyTwo,
+      providerId,
     );
-    if (dbUser) {
-      return dbUser;
+    if (authProvider) {
+      return this.userService.retrieveUserWithId(authProvider.userId);
     }
 
     const avatarDto = await this.saveAvatar(accessToken, avatarUrl);
-    return this.createUser(avatarDto, userDto);
+    const user = await this.createUser(avatarDto, userDto);
```

Comment:
todo: Since we can have multiple auth providers, there could be a collision of the `username` or the `email`.
We could generate a random `username` (provider `username` + append `providerId` or an arbitrary string?) and allow duplicate emails.